internal/mode/static/telemetry/collector.go

```diff
@@ -275,3 +263,56 @@ func CollectClusterID(ctx context.Context, k8sClient client.Reader) (string, err
 	}
 	return string(kubeNamespace.GetUID()), nil
 }
+
+// CollectNodeCount returns the number of nodes in the cluster.
+func CollectNodeCount(ctx context.Context, k8sClient client.Reader) (int, error) {
+	var nodes v1.NodeList
+	if err := k8sClient.List(ctx, &nodes); err != nil {
+		return 0, fmt.Errorf("failed to get NodeList: %w", err)
```

Comment:
Can we add a test case here?

Reply:
Yea I saw this, though this is covered in this part of the collector_test.go. 

```
It("should error on kubernetes client api errors", func() {
	expectedError := errors.New("failed to get NodeList")
	k8sClientReader.ListCalls(
		func(_ context.Context, object client.ObjectList, _ ...client.ListOption) error {
			switch object.(type) {
			case *v1.NodeList:
				return expectedError
			}
			return nil
		})

	_, err := dataCollector.Collect(ctx)
	Expect(err).To(MatchError(expectedError))
})
```

Since we call `k8sClient.List` on the NodeList twice and it should return the same thing I think it is fine as it will error out in `collectClusterInformation`. I think I may be able to spaghetti code some way with using the number of calls to specify what happens if it fails on the CollectNodeCount part but I don't think it would be very beneficial. Do you have any thoughts?

Reply:
I missed this during last review. But NodeCount fits ClusterInfo really well - can we add NodeCount field here:
```
type clusterInformation struct {
	Platform  string
	Version   string
	ClusterID string
        NodeCount int
}
```
this way we can get rid of CollectNodeCount in Collect all together. 

If this function isn't used anymore, we can move it to usage report package and make it private there.